### ISSUE TITLE
Optimize spacing sniff to check for array index if it is variable

### ISSUE THREAD

**Issue by shadyvb** [WordPress-Coding-Standards/WordPress-Coding-Standards#156]
As per [WPCS](http://make.wordpress.org/core/handbook/coding-standards/php/#space-usage):

> When referring to array items, only include a space around the index if it is a variable, for example:
> 
> ``` php
> $x = $foo['bar']; // correct
> $x = $foo[ 'bar' ]; // incorrect
> $x = $foo[0]; // correct
> $x = $foo[ 0 ]; // incorrect
> $x = $foo[ $bar ]; // correct
> $x = $foo[$bar]; // incorrect
> ```
